src/Kiota.Builder/Writers/CSharp/CodeMethodWriter.cs

```diff
@@ -259,14 +259,16 @@ private void WriteRequestGeneratorBody(CodeMethod codeElement, RequestParams req
                         $"PathParameters = {GetPropertyCall(urlTemplateParamsProperty, "string.Empty")},");
         writer.DecreaseIndent();
         writer.WriteLine("};");
+        if(codeElement.AcceptedResponseTypes.Any())
```

Comment:
So, if the `AcceptedResponseTypes` has both `application/json` and `text/xml` then the response content type could be any of the 2, right? Maybe we should use response content type in the paging service instead of the request Accept header value in this case.

Reply:
right now Kiota builder only ever sets one type, because it comes from the response media type in the schema. I'm using a hashet over a simple string to anticipate future evolutions (In case some APIs return the same schema with different media types and we want to generate only one method to deduplicate instead of one for each format)

Reply:
yes, unless we start sending priorities (q=) as well.
in the context of paging trying to parse the pages for other readers (#1572 ), it'd be better to rely on the content type of the response IMHO indeed.